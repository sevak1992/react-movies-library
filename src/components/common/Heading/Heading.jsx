import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles(() => ({
  heading: {
    marginTop: "2rem",
    marginBottom: "0.5rem",
    textTransform: "uppercase",
  },
}));

function Heading({ text, variant, color, component }) {
  const classes = useStyles();
  return (
    <Typography
      className={classes.heading}
      variant={variant}
      color={color}
      component={component}
    >
      {text}
    </Typography>
  );
}

Heading.propTypes = {
  text: PropTypes.string.isRequired,
  variant: PropTypes.string,
  color: PropTypes.string,
  component: PropTypes.string,
};

Heading.defaultProps = {
  variant: "h5",
  color: "textPrimary",
  component: "h4",
};

export default Heading;
