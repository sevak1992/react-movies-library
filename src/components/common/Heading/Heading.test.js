import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";

import Heading from "./Heading";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Heading component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Heading text="test 1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders with a text prop", () => {
    act(() => {
      render(<Heading text="test 1" />, container);
    });
    expect(container.textContent).toBe("test 1");

    act(() => {
      render(<Heading text="test 2" />, container);
    });
    expect(container.textContent).toBe("test 2");
  });
});
