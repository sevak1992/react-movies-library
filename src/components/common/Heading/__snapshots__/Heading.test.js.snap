// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading compomnent renders correctly 1`] = `
<h4
  className="MuiTypography-root makeStyles-heading-1 MuiTypography-h5 MuiTypography-colorTextPrimary"
>
  test 1
</h4>
`;

exports[`Heading component renders correctly 1`] = `
<h4
  className="MuiTypography-root makeStyles-heading-1 MuiTypography-h5 MuiTypography-colorTextPrimary"
>
  test 1
</h4>
`;
